okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -144,6 +144,8 @@ public void enqueue(Callback callback) {
         super.enqueue(new Callback() {
             @Override
             public void onFailure(Call call, IOException exception) {
+                urls.markFailed(request().url());
```

Comment:
what about 503s, do they count as "failed" in this sense?

Reply:
As is, yes. Happy to discuss if that's too broad of a failure condition

Reply:
I think the behavior you chose is actually the right one: markFailed on IOExceptions (indicating that the host is down), but not on 503 (which often indicates a temporary problem). Worth explaining this rationale in a comment though.